### PR TITLE
Add SSL support to astral api server

### PR DESCRIPTION
**Pull request by suprjinx**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -55,4 +55,4 @@ USER rails:rails
 # Start the server by default, this can be overwritten at runtime
 EXPOSE 3000
 # Start the server
-CMD ["bin/rails", "server", "-b", "0.0.0.0"]
\ No newline at end of file
+CMD ["bin/http.sh"]
```
Comment (GeorgeJahad) [inline Dockerfile:58]: Is this change just to make debugging easier?

Comment (GeorgeJahad) [reply Dockerfile:58]: oh, i'm guessing it is to be able to copy over it with ssh.sh?

File: bin/ssl.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#! /bin/sh
+SSL_CERT="${SSL_CERT:-cert/astral.pem}"
```
Comment (GeorgeJahad) [inline bin/ssl.sh:2]: Should we call these "ASTRAL_SSL_CERT/KEY" just to distinguish them from the other env vars?

File: lib/tasks/configure.rake [hunk 3]
```diff
@@ -2,22 +2,35 @@ require "rake"
...
 # Rake tasks for making a vault cert
 namespace :configure do
+  desc "Make Vault, Astral, and OIDC provider certs"
+  task ssl: [ :vault_ssl, :astral_ssl, :oidc_provider_ssl ]
+
   desc "Make the server cert for vault"
-  task :ssl, [ :cert_name ] do |t, args|
-    cert_name = args[:cert_name]
-    cert_name = "vault" if cert_name.nil?
-    sanParam = "subjectAltName=DNS:#{cert_name}"
```
Comment (GeorgeJahad) [inline lib/tasks/configure.rake:9]: The sanParam is actually needed for the oidc provider cert.  I mentioned it in my PR here: https://github.com/G-Research/astral/pull/49#issuecomment-2427657044

File: Dockerfile (hunk 1 above)
Comment (suprjinx) [reply Dockerfile:58]: yes, the ssl config involves some env vars, etc, so this made it simpler -- just did the same with http

File: lib/tasks/configure.rake (hunk 3 above)
Comment (suprjinx) [reply lib/tasks/configure.rake:9]: will restore -- I assume can we use this in all cases?

File: bin/ssl.sh (hunk 2 above)
Comment (suprjinx) [reply bin/ssl.sh:2]: good idea, will rename

File: lib/tasks/configure.rake (hunk 3 above)
Comment (GeorgeJahad) [reply lib/tasks/configure.rake:9]: > I assume can we use this in all cases?

That is my understanding.  I did recreate the vault cert with it, and all the tests still pass with the address set like so: "VAULT_ADDR=https://vault:8443"

**Review by GeorgeJahad — APPROVED** [review]
lgtm

**Pull request merged**